pkg/api/handlers/compat/containers_create.go

```diff
@@ -52,6 +52,13 @@ func CreateContainer(w http.ResponseWriter, r *http.Request) {
 		return
 	}
 
+	imageName, err := utils.NormalizeToDockerHub(r, body.Config.Image)
```

Comment:
Just a thought (I didn’t review the PR otherwise at all!)

Several (but by no means all) instances might be a tiny bit shorter with a
```go
if err := utils.NormalizeOverwriteToDockerHub(r, &body.Config.Image); err != nil { … }
```

with the obvious wrapper
```go
func NormalizeOverwriteToDockerHub(r …, inPlace *string) error {
   v, err := NormalizeToDockerHub(r, *inPlace)
   if err != nil { return err }
   *inPlace = v
   return nil
}
```

---

(\*grumble\* about Go’s error handling ergonomics)